### PR TITLE
fix Missing filter in issues#150

### PR DESCRIPTION
**Pull request by TYTaO**
Use hubHash to distinguish resources like other controllers

### CODE REVIEW

File: pkg/spoke/controllers/finalizercontroller/appliedmanifestwork_finalize_controller.go [hunk 1]
```diff
@@ -48,6 +50,9 @@ func NewAppliedManifestWorkFinalizeController(
 	return factory.New().
 		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
 			accessor, _ := meta.Accessor(obj)
+			if !strings.HasPrefix(accessor.GetName(), hubHash) {
+				return ""
```
Comment (qiujian16) [inline pkg/spoke/controllers/finalizercontroller/appliedmanifestwork_finalize_controller.go]: we have a queueKeyFunc in helper that you can use. Also I think we should use WithFilteredInformersQueueKeyFunc

Comment (TYTaO) [reply pkg/spoke/controllers/finalizercontroller/appliedmanifestwork_finalize_controller.go]: > 
Is the queueKeyFunc in helper this func？
https://github.com/open-cluster-management-io/work/blob/8f206974ddc0bb239e014768dbae19d0f37cad74/pkg/helper/helpers.go#L322
This func will return a manifestwork key, but in this controller we need a appliedmanifestwork key. May I can provide another func in helper?

Comment (qiujian16) [reply pkg/spoke/controllers/finalizercontroller/appliedmanifestwork_finalize_controller.go]: ok, that will be great. And maybe also a filter func in helper that can be reused by multiple controllers.

Comment (TYTaO) [reply pkg/spoke/controllers/finalizercontroller/appliedmanifestwork_finalize_controller.go]: > ok, that will be great. And maybe also a filter func in helper that can be reused by multiple controllers.

The changes have been committed

**Pull request merged**